### PR TITLE
framework/binary_manager : Split the function of setting request message

### PR DESCRIPTION
**Pull request by YiHyunJin**
…inary manager

There are many duplicated code.
It can be merged into defines for avoiding redundancy.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [0629deffb3888bee48c35f21141c60184871d782] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_internal.h:33: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/binary_manager/binary_manager_internal.h:33: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [0629deffb3888bee48c35f21141c60184871d782] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a18e22c5f6004dc5535adb81de507db00e405c22] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a18e22c5f6004dc5535adb81de507db00e405c22] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_internal.h:33: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/binary_manager/binary_manager_internal.h:33: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: framework/src/binary_manager/binary_manager_internal.h [hunk 1]
```diff
@@ -25,6 +25,18 @@
 
 #include <tinyara/binary_manager.h>
 
+#define SEND_REQUEST(request_msg, name, type)					\
+	do {									\
+		request_msg.cmd = type;						\
```
Comment (jeongchanKim) [inline framework/src/binary_manager/binary_manager_internal.h]: Please check alignment

File: framework/src/binary_manager/binary_manager_internal.h [hunk 2]
```diff
@@ -25,6 +25,18 @@
...
+	    request_msg.requester_pid = getpid();				\
```
Comment (jeongchanKim) [inline framework/src/binary_manager/binary_manager_internal.h]: Please add unistd.h

File: framework/src/binary_manager/binary_manager_internal.h (hunk 1 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: Thanks.

**Comment by seinfra** [comment]
Target : [4505d4c06b4b9bb5716f0e43c43d4d01ea7b721d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4505d4c06b4b9bb5716f0e43c43d4d01ea7b721d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5b72a2eb74be6ef301b16773add264019c38feb9] - Code Rule Check OK.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 2 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: I added

**Comment by seinfra** [comment]
Target : [5b72a2eb74be6ef301b16773add264019c38feb9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [35d346f67d73549b1d1c4966a05513fe09c1e273] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [35d346f67d73549b1d1c4966a05513fe09c1e273] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e705edde3cadb56cbc52c884c8f0e54d8c3c3bdd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e705edde3cadb56cbc52c884c8f0e54d8c3c3bdd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1d539d8c08c4359cdd1529a9bf9b89b7283a06ef] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1d539d8c08c4359cdd1529a9bf9b89b7283a06ef] - Code Rule Check OK.

File: framework/src/binary_manager/binary_manager_internal.h [hunk 3]
```diff
@@ -25,6 +25,18 @@
 
 #include <tinyara/binary_manager.h>
 
+#define SEND_REQUEST(type)								\
+	do {										\
+		request_msg.cmd = (type);						\
+		request_msg.requester_pid = getpid();					\
+		snprintf(request_msg.data.bin_name, BIN_NAME_MAX, "%s", binary_name);	\
```
Comment (sunghan-chang) [inline framework/src/binary_manager/binary_manager_internal.h]: include the header for snprintf

File: framework/src/binary_manager/binary_manager_internal.h [hunk 4]
```diff
@@ -25,6 +25,18 @@
 
 #include <tinyara/binary_manager.h>
 
+#define SEND_REQUEST(type)								\
+	do {										\
+		request_msg.cmd = (type);						\
+		request_msg.requester_pid = getpid();					\
```
Comment (sunghan-chang) [inline framework/src/binary_manager/binary_manager_internal.h]: header for getpid

File: framework/src/binary_manager/binary_manager_internal.h [hunk 5]
```diff
@@ -25,6 +25,18 @@
...
+		snprintf(request_msg.data.bin_name, BIN_NAME_MAX, "%s", binary_name);	\
+		ret = binary_manager_send_request(&request_msg);			\
+		if (ret < 0) {								\
+			bmdbg("Failed to send request msg %d\n", ret);			\
```
Comment (sunghan-chang) [inline framework/src/binary_manager/binary_manager_internal.h]: header for bmdbg

File: framework/src/binary_manager/binary_manager_internal.h [hunk 6]
```diff
@@ -25,6 +25,18 @@
 
 #include <tinyara/binary_manager.h>
 
+#define SEND_REQUEST(type)								\
+	do {										\
+		request_msg.cmd = (type);						\
+		request_msg.requester_pid = getpid();					\
+		snprintf(request_msg.data.bin_name, BIN_NAME_MAX, "%s", binary_name);	\
+		ret = binary_manager_send_request(&request_msg);			\
```
Comment (sunghan-chang) [inline framework/src/binary_manager/binary_manager_internal.h]: header for binary_manager_send_request

**Comment by seinfra** [comment]
Target : [0e2003368e79b2c5925dc9b63cd657d958b37fb5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0e2003368e79b2c5925dc9b63cd657d958b37fb5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4341d39467d7102eb9f7f0e1647e01eeca48c69f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4341d39467d7102eb9f7f0e1647e01eeca48c69f] - Code Rule Check OK.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 6 above)
Comment (sunghan-chang) [reply framework/src/binary_manager/binary_manager_internal.h]: check again

File: framework/src/binary_manager/binary_manager_update.c [hunk 7]
```diff
@@ -20,7 +20,6 @@
  * Included Files
  ***************************************************************************/
 
-#include <stdio.h>
 #include <stdlib.h>
```
Comment (sunghan-chang) [inline framework/src/binary_manager/binary_manager_update.c]: It looks many headers are unncessary.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 4 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: I added.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 3 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: I added.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 5 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: I added.

**Comment by seinfra** [comment]
Target : [6624ba029a85275706c980fdebc9e6011c32dd66] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6624ba029a85275706c980fdebc9e6011c32dd66] - Code Rule Check OK.

File: framework/src/binary_manager/binary_manager_internal.h (hunk 6 above)
Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_internal.h]: binary_manager_send_request is defined in this file. Therefore, there is no need to include header.

**Comment by seinfra** [comment]
Target : [e60de32fdbbb5875100439c55279dc4f90471561] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e60de32fdbbb5875100439c55279dc4f90471561] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [212a0ada84dafca5bac457512fdeaf426fc83987] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [212a0ada84dafca5bac457512fdeaf426fc83987] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [69e61479be29f07f21559f2343ddb1dafb848f54] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [69e61479be29f07f21559f2343ddb1dafb848f54] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a5c5b608a8fbe62ffe58ed0a65c651ca6a8baaf] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a5c5b608a8fbe62ffe58ed0a65c651ca6a8baaf] - Code Rule Check (C++) OK.

File: framework/src/binary_manager/binary_manager_interface.c [hunk 8]
```diff
@@ -24,10 +24,23 @@
...
 #include <fcntl.h>
 #include <mqueue.h>
 #include <tinyara/binary_manager.h>
 
+void binary_manager_set_request(binmgr_request_t *request_msg, int cmd, char *bin_name, binmgr_cb_t *callback)
+{
+	request_msg->cmd = cmd;
+	request_msg->requester_pid = getpid();
+	if (bin_name != NULL) {
+		snprintf(request_msg->data.bin_name, BIN_NAME_MAX, "%s", bin_name);
```
Comment (jeongchanKim) [inline framework/src/binary_manager/binary_manager_interface.c]: request_msg->data.bin_name and cb_info is union value. Please check that this is correct or not.

**Comment by seinfra** [comment]
Target : [a09e73c369fbf4cf09fa7ce60bb3b94cef48a78d] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_interface.c:41: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/binary_manager/binary_manager_interface.c:41: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

File: framework/src/binary_manager/binary_manager_interface.c [hunk 9]
```diff
@@ -24,10 +24,42 @@
...
+int binary_manager_set_request(binmgr_request_t *request_msg, int cmd, void *arg)
+{
+	if (request_msg == NULL) {
+		bmdbg("Invalid param\n");
+		return ERROR;
+	}
+	request_msg->cmd = cmd;
+	request_msg->requester_pid = getpid();
+
+	switch(cmd) {
```
Comment (jeongchanKim) [inline framework/src/binary_manager/binary_manager_interface.c]: Please add space after `switch`

File: framework/src/binary_manager/binary_manager_interface.c [hunk 10]
```diff
@@ -24,10 +24,42 @@
...
+				return ERROR;
+			}
+			snprintf(request_msg->data.bin_name, BIN_NAME_MAX, "%s", (char *)arg);
+			break;
+		case BINMGR_REGISTER_STATECB:
+			if (arg == NULL) {
+				bmdbg("Invalid param\n");
+				return ERROR;
+			}
+			request_msg->data.cb_info = (binmgr_cb_t *)arg;
```
Comment (jeongchanKim) [inline framework/src/binary_manager/binary_manager_interface.c]: Please add break.

**Comment by seinfra** [comment]
Target : [0d65a26c4d9e7c25db936fa1801f9c5d30b98e0b] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_interface.c:41: ERROR: [IDT_M_SCH] switch and case should be at the same indent

**Comment by seinfra** [comment]
Target : [0d65a26c4d9e7c25db936fa1801f9c5d30b98e0b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c50fadbadcb0e13b9ec59207fe71b7c7f8009599] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c50fadbadcb0e13b9ec59207fe71b7c7f8009599] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_interface.c:41: ERROR: [IDT_M_SCH] switch and case should be at the same indent

**Comment by seinfra** [comment]
Target : [d4575ead1728dddf026b94bfaa53411998099fe8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d4575ead1728dddf026b94bfaa53411998099fe8] - Code Rule Check Result:
framework/src/binary_manager/binary_manager_interface.c:41: ERROR: [IDT_M_SCH] switch and case should be at the same indent

File: framework/src/binary_manager/binary_manager_interface.c [hunk 11]
```diff
@@ -24,10 +24,43 @@
...
+			break;
+		case BINMGR_REGISTER_STATECB:
+			if (arg == NULL) {
+				bmdbg("Invalid param, cmd : %d\n", cmd);
+				return ERROR;
+			}
+			request_msg->data.cb_info = (binmgr_cb_t *)arg;
+			break;
+		default:
+		break;
```
Comment (hasw7569) [inline framework/src/binary_manager/binary_manager_interface.c]: Please add tab for break of default case as other cases

Comment (YiHyunJin) [reply framework/src/binary_manager/binary_manager_interface.c]: Thanks,

**Comment by seinfra** [comment]
Target : [96cec22c8a1278df6fe4c06ca72226b2710e39ac] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [96cec22c8a1278df6fe4c06ca72226b2710e39ac] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [021ec9e9c8b07f169679592ced3c29c2989b0d73] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [021ec9e9c8b07f169679592ced3c29c2989b0d73] - Code Rule Check OK.

**Pull request merged**